golem_messages/message/base.py

```diff
@@ -261,6 +262,8 @@ def serialize_slot(self, key, value):
             return value.value
         if key in self.MSG_SLOTS:
             return self.serialize_message(key, value)
+        if isinstance(value, TaskHeader):

```

Comment:
in `deserialize_slot()` you're checking againts a key (`if key == 'task_header' and value is not None`) IMO the same should be done here.

Reply:
checking if value is a `TaskHeader` instance is safer -> we know that we can call `value.to_dict()`; having just the key name and not being `None` doesn't give us the same level of certainty...

Reply:
This way you will silently serialize message with wrong field values, i.e. a `TaskHeader` in a `provider_public_key` slot.
Also I just noticed it's implemented in `Base`. IMO it should be implemented as a mixin in `golem_messages.datastructures.tasks`. Similar to `golem_messages.datastructures.p2p.NodeSlotMixin`.

Reply:
what about:
`if key == 'task_header' and isinstance(value, TaskHeader)` ? ;)

I will move it to 'WantToComputeTask'